File: app/src/main/java/com/example/plantonic/ui/activity/CheckoutActivityViewModel.java
package com.example.plantonic.ui.activity;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.example.plantonic.firebaseClasses.AddressItem;
import com.example.plantonic.repo.AddressRepository;
import com.example.plantonic.repo.BlueDartRepository;
import com.example.plantonic.retrofit.models.pincode.PinCodeAvailableResponseModel;

public class CheckoutActivityViewModel extends ViewModel {
    private AddressRepository addressRepository = new AddressRepository();

    public LiveData<AddressItem> currentAddress = addressRepository.userAddress;

    public void updateAddress(AddressItem addressItem){
        addressRepository.updateUserAddress(addressItem);
    }

    public void getAddress(String userId){
        addressRepository.getUserAddress(userId);
    }

    private  BlueDartRepository repo = new BlueDartRepository();
    public LiveData<Boolean> isLoading = repo.isLoading();
    public LiveData<String> errorMessage = repo.getErrorMessage();

    public LiveData<PinCodeAvailableResponseModel> isPinCodeAvailable = repo.isPinCodeAvailable();

    public void checkIfPinCodeAvailable(String pincode){
        repo.checkIsPinCodeAvailable(pincode);
    }
    
    
}

File: app/src/main/java/com/example/plantonic/ui/activity/CheckoutActivityViewModel.kt
package com.example.plantonic.ui.activity

import androidx.lifecycle.ViewModel
import com.example.plantonic.repo.AddressRepository
import androidx.lifecycle.LiveData
import com.example.plantonic.firebaseClasses.AddressItem
import com.example.plantonic.repo.BlueDartRepository

class CheckoutActivityViewModel : ViewModel() {
    private val addressRepository = AddressRepository()
    var currentAddress: LiveData<AddressItem> = addressRepository.userAddress
    fun updateAddress(addressItem: AddressItem?) {
        addressRepository.updateUserAddress(addressItem)
    }

    fun getAddress(userId: String?) {
        addressRepository.getUserAddress(userId)
    }

    private val repo = BlueDartRepository()
    val isLoading: LiveData<Boolean> = repo.isLoading
    val errorMessage: LiveData<String> = repo.errorMessage
    val isPinCodeAvailable = repo.isPinCodeAvailable
    fun checkIfPinCodeAvailable(pincode: String?) {
        repo.checkIsPinCodeAvailable(pincode!!)
    }
}